pandas/tests/series/indexing/test_setitem.py

```diff
@@ -740,6 +741,17 @@ def test_index_putmask(self, obj, key, expected, val):
 @pytest.mark.parametrize(
     "obj,expected,key",
     [
+        pytest.param(
+            # GH#45568 setting a valid NA value into IntervalDtype[int] should
+            #  cast to IntervalDtype[float]
+            Series(interval_range(1, 5)),
+            Series(
+                [Interval(1, 2), np.nan, Interval(3, 4), Interval(4, 5)],
```

Comment:
Is it worth testing `None` too here based on the whatsnew note?

Reply:
This is in the constructor, not in `__setitem__`.  The value being set with `__setitem__` is defined in the `TestSetitemCastingEquivalents.val` fixture, which includes None.